### DISCUSSION TITLE
Album sharing with your Cloudflare tunnel webaddress

### DISCUSSION THREAD

**Discussion by cazure-n00b** [photoprism/photoprism#4408] [Q&A]
Hi, I've currently setup my Photoprism through a Cloudflare Tunnel.

Now I want to share albums / folders with friends and family through my Cloudflare tunnel, but when I create a sharelink, it will use my IP address and port, after that, friend - of course - can't access it properly.

I must say I'm kinda lost on how to set it up correctly inside Photoprism / Docker Container (running it on my NAS), and I don't want to flunk it, so therefore. I'm here asking for help.

If someone could maybe guide me through the process and where to look, that would be awesome!

**Comment by graciousgrey** [comment]
Hi :wave: 

Unfortunately I cannot help with cloudflare but do you have the [PHOTOPRISM_SITE_URL](https://docs.photoprism.app/getting-started/config-options/) variable set to the url you use to access PhotoPrism?

**Reply by cazure-n00b** [reply]
Thanks for your reply!
I tried changing it, but Docker won't let me do it. It seems I'm not allowed to save it as an admin or whatsoever. Should I also drop the database (as in, stop it, make the changes, and then start both)?

**Reply by graciousgrey** [reply]
To change a environment variable you can stop the docker container (docker compose stop). Edit the docker-compose.yml file and restart the container (docker compose up).

**Reply by cazure-n00b** [reply]
Hi, sorry for my long absence.

I just did a re-install of Photoprism in Portainer. There I changed the PHOTOPRISM_SITE_URL to my Site URL: https://MY-WEBSITE-NAME.XYZ

I do run it through Cloudflare Tunnels. And there seems something to be going wrong. 
Going to: https://LOCALHOST:2342 works fine. But with cloudflare (tunnels), no luck.

Maybe someone has a clue on how to setup the final things.
Also did in Cloudflare a setup with http and https, but no luck (yet). See image attached.
![firefox_3e4PsiraL6](https://github.com/user-attachments/assets/dd778ccb-5d2e-4ead-83d1-9ab626e96340)

**Comment by DJKeterson** [comment]
You should first set it up without CF Tunnels and stick to the regular CF DNS zone. When you get that right and understand how it works it will be easier to set it up with tunnels. With limited info you shared it is hard to assist you with specific details.
What is the reason to use tunnels and not the regular DNS zone, assuming that security is your concern?

**Reply by cazure-n00b** [reply]
@DJKeterson in the research I did, I figured it would be the easiest, simplest way to work with CF Tunnels > CF DNS. 
As it was advertised as a more safer option in regarding - on a simple manner - not making anything open to the internet. Like the ports on your home network.

Is there more documentation you would suggest to read through regarding CF DNS Zone(s)?
Thanks in advance!

**Reply by DJKeterson** [reply]
@cazure-n00b I see but I don't mind having an opet port and for those who strive for security you can put it behind an IP ACL.

In short, you will need to forward port 443 to 2342 on your router towards your machine.

After that you should have a domain name (can be free) and simple free dyndns service of your choice (whichever is available in your router) so once you configure it your router will always update the dyndns record with current public IP.

Finally, you will need a cname record in your cloudflare pointing to your dyndns hostname as seen below, can be proxied.

![image](https://github.com/user-attachments/assets/056d98eb-95eb-40f2-8b34-f2b0b739f5a6)

SSL/TLS settings under cloudflare I've set to Flexible due to many other things running but you can try any option and see how it works, essentially this will get your PhotoPrism instance behind trusted certificate and allow encryption of data in transit.
